metaflow/extension_support.py

```diff
@@ -445,56 +619,80 @@ def _get_extension_packages():
                                 % (_extension_points[idx], config_module)
                             )
-                            packages_to_add.add(tl_fullname)
-                        else:
-                            # Check what directories we need to go down if any
-                            if len(ext_list) > 1 and prefix_match == ext_list[:-1]:
-                                if next_dirs is None:
-                                    next_dirs = []
-                                next_dirs.append(ext_list[-1])
-                    if next_dirs is not None:
-                        dirs[:] = next_dirs[:]
-
-        # Add all these new packages to the list of packages as well.
-        packages_to_add = list(packages_to_add)
-        packages_to_add.sort()
-        mf_pkg_list.extend(packages_to_add)
+            mf_pkg_list.append(package_name)
+            mf_ext_packages[package_name] = {
+                "root_paths": [package_path],
+                "meta_module": meta_module,
+                "files": files_to_include,
+            }
 
     # Sanity check that we only have one package per configuration file
     errors = []
     for m, packages in config_to_pkg.items():
         if len(packages) > 1:
             errors.append(
                 "\tPackages %s define the same configuration module '%s'"
-                % (", and ".join(packages), m)
+                % (", and ".join(["'%s'" % p for p in packages]), m)
+            )
+    for m, packages in meta_to_pkg.items():
+        if len(packages) > 1:
+            errors.append(
+                "\tPackages %s define the same meta module '%s'"
+                % (", and ".join(["'%s'" % p for p in packages]), m)
             )
     if errors:
         raise RuntimeError(
-            "Conflicts in %s configuration files:\n%s" % (EXT_PKG, "\n".join(errors))
+            "Conflicts in '%s' files:\n%s" % (EXT_PKG, "\n".join(errors))
         )
 
     extension_points_to_pkg.default_factory = None
     # Figure out the per extension point order
     for k, v in extension_points_to_pkg.items():
+        # v is a dict distributionName/packagePath -> (dict tl_name -> MFPackage)
         l = [v[pkg].values() for pkg in mf_pkg_list if pkg in v]
-        extension_points_to_pkg[k] = list(chain(*l))
-    return mf_pkg_list, extension_points_to_pkg
+        # In the case of the plugins.cards extension, we allow those packages
+        # to be ns packages so we only list the package once (in its first position).
+        # In all other cases, we error out if we don't have a configuration file for the
+        # package (either a __init__.py of an explicit mfextinit_*.py)
+        final_list = []
+        have_null_config = False
+        for pkg in chain(*l):
+            if pkg.config_module is None:
+                if k == "plugins.cards":
+                    # This is allowed here but we only keep one
+                    if have_null_config:
+                        continue
```

Comment:
When I was testing, I noticed that we are swallowing all other card-related modules and explicitly keeping only one. Below is the state of variable `extension_points_to_pkg` at time of for loop. 
```python
defaultdict(None, {'plugins.cards': {'metaflow-card-uplot': {'card_uplot': MFExtPackage(package_name='metaflow-card-uplot', tl_package='card_uplot', config_module=None)}, 'metaflow-card-notebook': {'card_notebook': MFExtPackage(package_name='metaflow-card-notebook', tl_package='card_notebook', config_module=None)}, 'metaflow-card-html': {'card_html': MFExtPackage(package_name='metaflow-card-html', tl_package='card_html', config_module=None)}}})
```
Due to the `continue` we only take the first package and ignore the rest since we don't add them to `final_list`.